compaction/compaction_manager.cc

```diff
@@ -1785,6 +1785,15 @@ compaction_backlog_tracker::compaction_backlog_tracker(compaction_backlog_tracke
         , _manager(std::exchange(other._manager, nullptr)) {
 }
 
+compaction_backlog_tracker&
+compaction_backlog_tracker::operator=(compaction_backlog_tracker&& x) noexcept {
+    if (this != &x) {
+        this->~compaction_backlog_tracker();
+        new (this) compaction_backlog_tracker(std::move(x));
```

Comment:
Please avoid this new (this) style in new code, I'm not sure it's totally legal.

Reply:
In particular, if *this has a const member (directly or indirectly), the compiler is allowed to assume it doesn't change, but new (this) can change it. Maybe.